pkg/operator/workload_controller_openshiftapiserver_v311_00.go

```diff
@@ -89,49 +89,95 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
+	case actualDaemonSet.Status.NumberAvailable > 0 && len(actualAPIServices) == 0:
+		availableConditionStatus = operatorv1.ConditionUnknown
+		availableConditionReason = "APIServiceAvailabilityUnknown"
+		availableConditionMessages = append(availableConditionMessages, "apiservice/*: status unknown")
+	}
+
+	if actualDaemonSet != nil && actualDaemonSet.Status.NumberAvailable == 0 {
+		availableConditionReason = "NoAPIServerPod"
+		availableConditionMessages = append(availableConditionMessages, "no openshift-apiserver daemon pods available on any node.")
+	}
+	for _, apiService := range actualAPIServices {
+		for _, condition := range apiService.Status.Conditions {
+			if condition.Type == apiregistrationv1.Available {
+				if condition.Status == apiregistrationv1.ConditionFalse {
+					availableConditionReason = "APIServiceNotAvailable"
+					availableConditionMessages = append(availableConditionMessages, fmt.Sprintf("apiservice/%v: not available: %v", apiService.Name, condition.Message))
+				}
+				break
+			}
+		}
+	}
+	switch {
+	case len(availableConditionMessages) == 1:
 		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
-			Type:   operatorv1.OperatorStatusTypeAvailable,
-			Status: operatorv1.ConditionTrue,
+			Type:    operatorv1.OperatorStatusTypeAvailable,
+			Status:  availableConditionStatus,
+			Reason:  availableConditionReason,
+			Message: availableConditionMessages[0],
 		})
-	} else {
+	case len(availableConditionMessages) > 1:
 		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
 			Type:    operatorv1.OperatorStatusTypeAvailable,
-			Status:  operatorv1.ConditionFalse,
-			Reason:  "NoPodsAvailable",
-			Message: "no daemon pods available on any node.",
+			Status:  availableConditionStatus,
+			Reason:  "Multiple",
+			Message: strings.Join(availableConditionMessages, "\n"),
+		})
+	default:
+		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
+			Type:   operatorv1.OperatorStatusTypeAvailable,
+			Status: operatorv1.ConditionTrue,
 		})
 	}
 
 	// If the daemonset is up to date and the operatorConfig are up to date, then we are no longer progressing
-	var progressingMessages []string
-	if actualDaemonSet.ObjectMeta.Generation != actualDaemonSet.Status.ObservedGeneration {
-		progressingMessages = append(progressingMessages, fmt.Sprintf("daemonset/apiserver.openshift-operator: observed generation is %d, desired generation is %d.", actualDaemonSet.Status.ObservedGeneration, actualDaemonSet.ObjectMeta.Generation))
-	}
-	if operatorConfig.ObjectMeta.Generation != operatorConfig.Status.ObservedGeneration {
-		progressingMessages = append(progressingMessages, fmt.Sprintf("openshiftapiserveroperatorconfigs/instance: observed generation is %d, desired generation is %d.", operatorConfig.Status.ObservedGeneration, operatorConfig.ObjectMeta.Generation))
-	}
-
-	if len(progressingMessages) == 0 {
```

Comment:
why did you touch this

Reply:
to avoid panic when `actualDaemonSet == nil`